var/spack/repos/builtin/packages/openspeedshop/package.py

```diff
@@ -144,19 +153,52 @@ class Openspeedshop(CMakePackage):
     depends_on('cbtf-krell+openmpi', when='+cbtf+openmpi')
     depends_on("cbtf-argonavis", when='+cbtf+cuda')
     depends_on("mrnet@5.0.1:+lwthreads", when='+cbtf')
+    depends_on("mrnet@5.0.1:+cti", when='+cti+cbtf')
 
     parallel = False
 
     build_directory = 'build_openspeedshop'
 
+    def set_CrayLoginNode_cmakeOptions(self, spec, cmakeOptions):
+        # Appends to cmakeOptions the options that will enable the appropriate
+        # Cray login node libraries
+
+        CrayLoginNodeOptions = []
+        rt_platform = "cray"
+        # How do we get the compute node (CNL) cbtf package install
+        # directory path?
+        # spec['cbtf'].prefix is the login node value for this build, as
+        # we only get here when building the login node components and
+        # that is all that is known to spack.
+        be_ck = spack.store.db.query_one('cbtf-krell arch=cray-CNL-haswell')
+
+        # Equivalent to install-tool cmake arg:
+        # '-DCBTF_KRELL_CN_RUNTIME_DIR=%s'
+        #               % <base dir>/cbtf_v2.3.1.release/compute)
+        CrayLoginNodeOptions.append('-DCBTF_KRELL_CN_RUNTIME_DIR=%s'
+                                    % be_ck.prefix)
+        CrayLoginNodeOptions.append('-DRUNTIME_PLATFORM=%s'
+                                    % rt_platform)
+
+        cmakeOptions.extend(CrayLoginNodeOptions)
+
     def cmake_args(self):
+
         spec = self.spec
+
+        # Checking the platform and target
+        # OpenSpeedShop should not be built for the compute nodes
+        # when building in cbtf mode (+cbtf)
+        # platform = self.spec.architecture.platform
+        # if spec.satisfies('~offline'):
```

Comment:
@jgalarowicz Is this commented code left intentionally? If so it may be worthwhile to mention why it is left here.

Reply:
@scheibelp I couldn't get it to work properly, but I want to flag that openspeedshop should not be built for the compute nodes.

Reply:
@jgalarowicz Ah ok - the platform can be `cray` on both the front and backend nodes. `conflicts('os=be')` may work. How was it failing?

Reply:
@scheibelp spack was complaining that I was defining platform more than once.  I can try your suggestion.  Is "be" a generic term in spack for backend builds?

Reply:
@jgalarowicz Yes - it should be substituted with the backend os on the platform (hopefully this is quick to test since the goal is to get it to refuse to build)

Reply:
I took the comments out and will have to work on a solution for the issue later.  It is not as important as getting this merged.    It didn't catch the conflicts statement and just started building OSS for the compute nodes.